### PR TITLE
Allow THEME_DIR to be customised

### PR DESCRIPTION
**Pull request by wilr**
Adds flexibility for custom setups such as app/themes to be implemented in the case a project wants to provide a custom path.

### CODE REVIEW

**Comment by Zauberfisch** [comment]
good to merge.

@wilr though have you searched for hard coded references to `themes/`? I wouldn't be surprised to find a few across framework and cms.

**Comment by tractorcow** [comment]
@wilr you pushed your wip branch directly to origin. :)

File: core/Constants.php [hunk 1]
```diff
@@ -229,10 +229,19 @@ function stripslashes_recursively(&$array) {
 		define('BASE_URL', rtrim($dir, DIRECTORY_SEPARATOR));
 	}
 }
-define('MODULES_DIR', 'modules');
+
+if (!defined('MODULES_DIR')) {
+	define('MODULES_DIR', 'modules');
```
Comment (tractorcow) [inline core/Constants.php]: We have NEVER respected this constant before... I'm surprised to see it actually.

**Comment by tractorcow** [comment]
I think you'll need to fix hard-coded themes paths before getting this over the line... I can see a few.

- i18n.php
- i18nTextCollector.php

:D

**Comment by tractorcow** [comment]
I've backed up your work to https://github.com/tractorcow/silverstripe-framework/tree/wilr-custom-theme-dir... I'll remove your branch and let you create another PR from your fork. Is that ok?

**Comment by wilr** [comment]
Didn't we decide to implement merge by default @tractorcow for PRs in the last core chat.. Simple change that had no side effects. Now requires more effort.. #justsaying

**Comment by tractorcow** [comment]
Yeah but there are side effects if you change themes path and there are hard coded references. :)

**Pull request closed without merging**